src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -2100,6 +2112,22 @@ - (void)unitTestClearClusterData
 }
 #endif
 
+- (void)_reconcilePersistedClustersWithStorage
+{
+    os_unfair_lock_assert_owner(&self->_lock);
+
+    NSMutableSet * clusterPathsToRemove = [NSMutableSet set];
+    for (MTRClusterPath * clusterPath in _persistedClusters) {
+        MTRDeviceClusterData * data = [_deviceController.controllerDataStore getStoredClusterDataForNodeID:_nodeID endpointID:clusterPath.endpoint clusterID:clusterPath.cluster];
+        if (!data) {
+            [clusterPathsToRemove addObject:clusterPath];
+        }
+    }
+
+    MTR_LOG_ERROR("%@ ", self);
```

Comment:
Is this safe, also what is this logging?

Reply:
Meant to log the difference between in-memory cache and storage - I've fixed it now. (And also the incomplete log was safe, even if unclear)